google/cloud/spanner/sql_statement.cc

```diff
@@ -19,15 +19,35 @@ namespace google {
 namespace cloud {
 namespace spanner {
 inline namespace SPANNER_CLIENT_NS {
+
+std::vector<std::string> SqlStatement::ParameterNames() const {
+  std::vector<std::string> keys;
+  keys.reserve(params_.size());
+  for (auto const& p : params_) {
+    keys.push_back(p.first);
+  }
+  return keys;
+}
+
+google::cloud::StatusOr<Value> SqlStatement::GetParameter(
+    std::string const& parameter_name) const {
+  auto iter = params_.find(parameter_name);
+  if (iter != params_.end()) {
+    return iter->second;
+  }
+  return Status(StatusCode::kNotFound, "No such parameter.");
```

Comment:
Optional: I'd suggest putting `parameter_name` in the error message.

Reply:
Done